File: Problems/Food Delivery System/Java/payments/OnlinePaymentStrategy.java
package payments;

public class OnlinePaymentStrategy implements Payment {
    @Override
    public boolean pay(double amount) {
        // Logic for online payment
        System.out.println("Processing online payment of: " + amount);
        // Assume payment is successful
        System.out.println("Online payment successful.");
        return true;
    }
}

File: Problems/Food Delivery System/Golang/payments/online_payment_strategy.go
package payments

import "fmt"

// OnlinePaymentStrategy implements Payment using an online gateway (simulated).
type OnlinePaymentStrategy struct{}

func NewOnlinePaymentStrategy() *OnlinePaymentStrategy {
	return &OnlinePaymentStrategy{}
}

func (o *OnlinePaymentStrategy) Pay(amount float64) bool {
	fmt.Println("Processing online payment of:", amount)
	fmt.Println("Online payment successful.")
	return true
}
